aten/src/ATen/native/Unique.cpp

```diff
@@ -154,6 +154,28 @@ std::tuple<Tensor, Tensor, Tensor> _unique_dim_cpu_template(
     const bool consecutive,
     const bool return_inverse,
     const bool return_counts) {
+  // check if dim is in range and see if the size along that dim is 0
+  if (dim <= self.dim() && self.size(dim) == 0) {
+    // assume tensor has all dimentions non-zero
+    bool well_formed_tensor = true;
+    for (int64_t i = 0; i < self.dim(); ++i) {
+      if (dim == i)
+        continue;
+      if (self.size(i) == 0) {
```

Comment:
Would it be better to change the whole `if` to something like
```C++
AT_CHECK(self.size(i) > 0, "Some error message here");
```

Reply:
I don't think there is any document for these functions. The best way to learn may be looking at other codes that uses those functions...